numpy/_core/numerictypes.py

```diff
@@ -359,6 +360,90 @@ def issubsctype(arg1, arg2):
     return issubclass(obj2sctype(arg1), obj2sctype(arg2))
 
 
+@set_module('numpy')
+def isdtype(dtype, kind):
+    """
+    Returns a boolean indicating whether a provided dtype is of a specified
+    data type ``kind``. This function only supports Array API data types.
+
+    Parameters
+    ----------
+    dtype : dtype
+        The input dtype.
+    kind : dtype or str or tuple of dtypes/strs.
+        dtype or dtype kind. Allowed dtype kinds are:
+        * ``bool`` : boolean kind
+        * ``signed integer`` : signed integer data types
+        * ``unsigned integer`` : unsigned integer data types
+        * ``integral`` : integer data types
+        * ``real floating`` : real-valued floating-point data types
+        * ``complex floating`` : complex floating-point data types
+        * ``numeric`` : numeric data types
+
+    Returns
+    -------
+    out : bool
+
+    See Also
+    --------
+    issubdtype
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> np.isdtype(np.float32, np.float64)
+    False
+    >>> np.isdtype(np.int64, (np.uint64, "signed integer"))
+    True
+
+    """
+    # validate and preprocess arguments
+    if isinstance(dtype, ma.dtype):
+        dtype = dtype.type
+
+    if dtype not in allTypes.values():
+        raise TypeError(
+            "input_dtype argument must be a NumPy dtype, "
+            f"but it is a {dtype}."
```

Comment:
I think this should be `{type(dtype)}`? Try with `np.isdtype('np.float32', 'real floating')`.

Reply:
Right, fixed.